extensions/amp-a4a/0.1/head-validation.js

```diff
@@ -0,0 +1,189 @@
+/**
+ * Copyright 2020 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Services} from '../../../src/services';
+import {map} from '../../../src/utils/object';
+import {parseExtensionUrl} from '../../../src/service/extension-location';
+import {removeElement} from '../../../src/dom';
+
+/**
+ * @typedef {{
+ *    extensions: !Array<{extensionId: (string|undefined), extensionVersion: (string|undefined)}>,
+ *    head: !Element
+ * }}
+ */
+export let ValidatedHeadDef;
+
+const FONT_ALLOWLIST = map({
```

Comment:
Do we want to use a Set for this and EXTENSION_ALLOWLIST?

Reply:
I would love to use set! Last I checked it was disallowed, has this changed?

Reply:
Looks like it was added in #26996 🎉

Reply:
hmm looks like you still can't pass an iterator to the constructor (ie `new Set(['amp-img', 'amp-video'])` ) so it winds up looking like 
```js
const EXTENSION_ALLOWLIST = new Set();
EXTENSION_ALLOWLIST.add('amp-accordion')
EXTENSION_ALLOWLIST.add('amp-ad-exit')
EXTENSION_ALLOWLIST.add('amp-analytics')
...
```
or 
```js
[ 
  'amp-accordion',
  'amp-ad-exit',
  'amp-analytics',
...
].forEach(ext => EXTENSION_ALLOWLIST.add(extension));
``` 
I'm not sure either is more readable. WDYT?

Reply:
`new Set(['amp-img', 'amp-video'])` should work ideally. Talked to justin, it is not allowed because we must polyfill for IE11. We should use object now.